### DISCUSSION TITLE
使用mysql cdc 全量快照+增量同步时，全量并发数远远大于增量时的单并发数，有没有回收这些资源的办法

### DISCUSSION THREAD

**Discussion by wangyusong001** [apache/flink-cdc#894] [Q&A]
全量同步快照时，指定4个并发同步，但这种操作往往进行一次，当到了增量同步时，会有3个slot长期空闲。任务数量比较多，我们想实现自动化的流程，那就会有下面的问题
问题：
1.是否有实现自动回收的能力
2.如果需要手动重启回收，有没有办法能判断全量快照已经完成

**Comment by GOODBOY008** [comment]
可以考虑将同步任务拆分成快照和增量两个单独的任务。

**Reply by wangyusong001** [reply]
如果拆成两个任务的话，使用CDC同步全量的意义就没有了。我之前可以用spark、sqoop等其他分布式的工具实现全量同步

**Comment by yangrong688** [comment]
可以通过脚本去监控是否完成全量，监控到完成增量后进行缩容

**Reply by yangrong688** [reply]
我们已经初步的通过 yarn log  的日志实现了一版，大致流程如下图所示
后续等 https://github.com/ververica/flink-cdc-connectors/pull/853 这个 pr 合并后可以去拉取 metric来获取任务是否完成snapshot阶段的状态
![flink_scale_flow](https://user-images.githubusercontent.com/70557521/157677681-354d7aa8-4d1b-4fff-952d-8d0a80bf8d74.png)

**Comment by nailding1** [comment]
使用多个实例生成全量索引，完毕之后保存状态停止，然后用少量实例从保存的状态启动job就行了
